connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginsTest.java

```diff
@@ -185,6 +189,68 @@ public void shouldInstantiateAndConfigureDefaultHeaderConverter() {
         assertTrue(headerConverter instanceof SimpleHeaderConverter);
     }
 
+    @Test(expected = ConnectException.class)
+    public void shouldThrowIfPluginThrows() {
+        TestPlugins.assertInitialized();
+
+        assertNotNull(plugins.newPlugin(
+            TestPlugins.ALWAYS_THROW_EXCEPTION,
+            new AbstractConfig(new ConfigDef(), Collections.emptyMap()),
+            Converter.class
+        ));
+    }
+
+    @Test(expected = ConnectException.class)
+    public void shouldCacheExceptionalClassInitialization() {
```

Comment:
Just out of curiosity, would you mind giving a brief explanation of what situations this test is meant to emulate that might require the exact behavior that's verified here? Or in other words, if this test failed for some reason, what might that shed light on as far as bugs in the code base go?

Reply:
This test is because of a "bug" i discovered while trying to add these tests. `plugins` used to be created once before all of the test cases, and so this caching effect was causing my negative test case to affect my positive test case. I included it as justification for creating a new instance of `plugins` for each test case.

This test would fail if the delegating classloader didn't implement the caching system, which would be extremely noticeable as a performance regression. I saw many other tests failing due to classloading performance issues, so this probably wouldn't be the only test to fail if such a bug was introduced.It would however be the only test that explicitly checks for that cause, whereas the other test failures are integration tests that time out, and aren't immediately traceable to this caching change.

Reply:
Hmm... I'd welcome input from others, but right now I'm leaning towards the opinion that this test might not be worth including here. If there isn't any reason to believe that that the caching behavior might be affected by changes to the `Plugins`, `DelegatingClassLoader`, `PluginClassLoader`, etc. classes now or in the future then the benefits of the test would only be limited to making sure nobody alters the test code in a way that would break things. A comment detailing why a new `Plugins` instance is created for each test (and/or how things would break if this didn't happen) seems just as useful and would add less cognitive burden on people reading becoming familiar with this part of the code base.

Reply:
I've removed this case since it seemed to be of marginal value, and isn't worth keeping an extra test plugin around for.